### ISSUE TITLE
Type check failed on static members

### ISSUE THREAD

**Issue by sirian** [Microsoft/TypeScript#29932]
**TypeScript Version:**  3.4.0-dev.20190215

**Code**

```ts
class Foo {
    public static foo() {
        return new Date();
    }
}

const x1: { foo(): Number } = Foo;
const x2: { foo(): Date } = Foo;
const x3: { foo(): Foo } = Foo;

class Bar extends Array {
    public static foo() {
        return new Date();
    }
}

const y1: { foo(): Number } = Bar;
const y2: { foo(): Date } = Bar;
const y3: { foo(): Bar } = Bar;
```

**Expected behavior:**
`const x3: { foo(): Foo } = Foo;` should trigger error `{foo: () => Date} is not assignable to {foo: () => Foo}` like `y3`

**Actual behavior:**
`const x3: { foo(): Foo } = Foo;` error not shown

[Playground Link](https://www.typescriptlang.org/play/index.html#src=class%20Foo%20%7B%0D%0A%20%20%20%20public%20static%20foo()%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20new%20Date()%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Aconst%20x1%3A%20%7B%20foo()%3A%20Number%20%7D%20%3D%20Foo%3B%0D%0Aconst%20x2%3A%20%7B%20foo()%3A%20Date%20%7D%20%3D%20Foo%3B%0D%0Aconst%20x3%3A%20%7B%20foo()%3A%20Foo%20%7D%20%3D%20Foo%3B%0D%0A%0D%0Aclass%20Bar%20extends%20Array%20%7B%0D%0A%20%20%20%20public%20static%20foo()%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20new%20Date()%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Aconst%20y1%3A%20%7B%20foo()%3A%20Number%20%7D%20%3D%20Bar%3B%0D%0Aconst%20y2%3A%20%7B%20foo()%3A%20Date%20%7D%20%3D%20Bar%3B%0D%0Aconst%20y3%3A%20%7B%20foo()%3A%20Bar%20%7D%20%3D%20Bar%3B%0D%0A)

**Comment by dragomirtitian** [comment]
@sirian  This is the expected behavior the instance type `Foo` has no members, it is equivalent to `{}` as far as the type checker is concerned. Typescript uses structural typing so if the structure of a type is compatible with another type the types are compatible.

So for example, given your class `Foo` with no members, these assignments are also valid:

```ts
const f1: Foo = {}
const f2: Foo = 2
const f3: Foo = "anything"
const f4: Foo = { literally: "anything"}
```

The reason you get an error when you add inheritance is that `Bar` will inherit members from `Array`, so it will no longer be the same as `{}`. You will also get an error in `Foo` as soon as you add any non-static member to the class:


```ts
class Foo {
    private m: any;
    public static foo() {
        return new Date();
    }
}
const x3: { foo(): Foo } = Foo; // error now
```
